### ISSUE TITLE
Unable to find matplotlib files in compiled version with numpy plugin

### ISSUE THREAD

**Issue by dhombios** [Nuitka/Nuitka#943]
I've tried compiling a tkinter app using Nuitka develop version (doe to a [previous error](https://github.com/Nuitka/Nuitka/issues/935) in the stable branch it is not possible to achieve that), but the resulting executable raises an exception due to being unable to find matplotlib data files

Platform: Windows (Anaconda)
Python: 3.7.9 (x32)
Nuitka: 0.6.11rc9

Compiler command:
`nuitka --follow-imports --standalone --plugin-enable=pylint-warnings --plugin-enable=tk-inter --plugin-enable=numpy --plugin-enable=qt-plugins main.py`

CMD output when executing the resulting executable:
```
D:\Dhom\Documents\University\IIC\SEA\interface\CompileV1\main.dist>main.exe
D:\Dhom\Documents\University\IIC\SEA\interface\CompileV1\main.dist\main.py:13: MatplotlibDeprecationWarning: Matplotlib installs where the data is not in the mpl-data subdirectory of the package are deprecated since 3.2 and support for them will be removed two minor releases later.
Traceback (most recent call last):
  File "D:\Dhom\Documents\University\IIC\SEA\interface\CompileV1\main.dist\main.py", line 13, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "D:\Dhom\Documents\University\IIC\SEA\interface\CompileV1\main.dist\matplotlib\__init__.py", line 914, in <module matplotlib>
  File "D:\Dhom\Documents\University\IIC\SEA\interface\CompileV1\main.dist\matplotlib\cbook\__init__.py", line 480, in _get_data_path
  File "D:\Dhom\Documents\University\IIC\SEA\interface\CompileV1\main.dist\matplotlib\__init__.py", line 239, in wrapper
  File "D:\Dhom\Documents\University\IIC\SEA\interface\CompileV1\main.dist\matplotlib\__init__.py", line 534, in get_data_path
  File "D:\Dhom\Documents\University\IIC\SEA\interface\CompileV1\main.dist\matplotlib\__init__.py", line 239, in wrapper
  File "D:\Dhom\Documents\University\IIC\SEA\interface\CompileV1\main.dist\matplotlib\__init__.py", line 566, in _get_data_path
RuntimeError: Could not find the matplotlib data files
```

**Comment by kayhayen** [comment]
Which version of matplotlib is this, I kind of remember having it working. I don't have Anaconda anymore, so it's hard to tell, take output from pip freeze there.

**Comment by dhombios** [comment]
It uses matplotlib 3.3.3 and its tkinter backend:

```
import tkinter as tk
from tkinter import ttk
from tkinter import messagebox as mbox
import matplotlib
matplotlib.use("TkAgg")
from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg, NavigationToolbar2Tk
from matplotlib.figure import Figure
```

**Comment by dhombios** [comment]
And the complete output from pip freeze is:
```
certifi==2020.12.5
cycler==0.10.0
enum34==1.1.10
kiwisolver==1.3.1
matplotlib==3.3.3
numpy==1.19.5
pandas==1.2.0
Pillow==8.1.0
pyparsing==2.4.7
pyserial==3.5
pytelemetry==1.1.9
python-dateutil==2.8.1
pytz==2020.5
six==1.15.0
tk==0.1.0
wincertstore==0.2
```

**Comment by kayhayen** [comment]
Ok, it seems they removed support for the environment variable that Nuitka has been using to make it find the `mpl-data` directory,  and instead require it to be properly located relative to `__file__` which is a good idea anyway. So the solution is to move that directory in the dist folder to where mpl-data lives.

That would make the pre-load code unnecessary, except it isn't, so I think, the matplotlib detection needs to gain some sort of discovery if the environment variable is in the source code or not. If it is not, we can avoid the pre-load code for matplotlib and be good.

**Comment by dhombios** [comment]
Thanks again, are there nightly builds of the factory branch that can be used to test it or are they just generated for the development branch?

**Comment by kayhayen** [comment]
There is this, but since this will be shortly in a 0.6.11.1 anyway, I didn't bother to point to it https://nuitka.net/doc/factory.html

**Comment by dhombios** [comment]
Thanks, I thought I would have to wait until the release of version 0.6.12

**Comment by kayhayen** [comment]
I tend to include contained fixes, esp. for important packages in a next hotfix, that way I am taking pressure of the next release to become available. So I develop stuff for a while on factory, and then split it out in a hotfix branch and a portion that goes into the next pre-release. 

On factory though, e.g. I did unary operator optimization and progres bar work, both of which might cause issues and are not yet final, so I am rebasing, and even this plugin fix might not be entirely final. Of course I had to add some cleanup that made the code more likely to be correct and regressed. :)
